core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/FileStateRepository.java

```diff
@@ -207,7 +207,10 @@ protected void loadStore() throws IOException {
             LOG.debug("Creating filestore: {}", fileStore);
             File parent = fileStore.getParentFile();
             if (parent != null) {
-                parent.mkdirs();
+                boolean mkdirsResult = parent.mkdirs();
+                if (!mkdirsResult) {
+                    LOG.error("mkdirs() failed for " + parent);
```

Comment:
My .2 cents: I think it would be better to have a more descriptive log message here since this is in the core of Camel and also using log markers. I'd suggest something like `LOG.error(""Couldn't create the filestore at {} because creating the directory has f" + parent);`. Alternatively, we could also explore whether [Files.createDirectories](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createDirectories-java.nio.file.Path-java.nio.file.attribute.FileAttribute...-) could fulfill this gaps as, I believe, it provides better error descriptions.